### ISSUE TITLE
'no-unused-vars' only effects parameters

### ISSUE THREAD

**Issue by DAmNRelentless** [eslint/eslint#11292]
**Tell us about your environment**

* **ESLint Version:** 1.8.0
* **Node Version:** 10.15.0
* **npm Version:** 6.4.1

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
<details>
<summary>Configuration</summary>

```js
{
    "env": {
        "browser": true,
        "es6": true,
        "node": true
    },
    "extends": "eslint:recommended",
    "parserOptions": {
        "ecmaVersion": 2018,
        "sourceType": "module"
    },
    "rules": {
        "indent": [
            "warn",
            2,
            {
              "SwitchCase": 1
            }
        ],
        "linebreak-style": [
            "warn",
            "unix"
        ],
        "quotes": [
            "warn",
            "single"
        ],
        "semi": [
            "warn",
            "never"
        ],
        "no-unused-vars": [
          "warn",
          {
            "args": "all"
          }
        ]
    }
}
```
</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**
I just coded my project and defined some variables that are not used yet and a function with parameters that are not used yet.

```js
// For each server in the config
async.eachOf(config.server, (serverConfig, serverName, callback) => {
  // Some config defaults
  serverConfig.queryport = serverConfig.queryport || '10011'
  serverConfig.serverport = serverConfig.serverport || '9987'
  serverConfig.nickname = serverConfig.nickname || 'Bot'
  serverConfig.username = serverConfig.username || 'serveradmin'

  const ts3 = new TeamSpeak3(serverConfig)
})
```

**What did you expect to happen?**
I set in my configuration that unused vars are marked as warnings and not as errors anymore.
I expect this to happen but this only effects the parameters that are passed to the function.
If I define a variable somewhere which is not used, it's still marked as error.

**What actually happened? Please include the actual, raw output from ESLint.**
It marked the ts3 variable as error and the parameters as warnings. I want them all to be only warnings.
![unbenannt](https://user-images.githubusercontent.com/16513358/51421204-35eb3a80-1b9b-11e9-8f87-8a1321618000.PNG)

**Are you willing to submit a pull request to fix this bug?**
No

**Comment by DAmNRelentless** [comment]
I am dumb, it comes from standard. I forgot to disable it.
